jenkins/debian/debos/scripts/crush.sh

```diff
@@ -1,156 +1,202 @@
 
-# systemd ntp client (connman is in use)
-#find usr etc -name '*timesyncd*' -prune -exec rm -r {} \;
+    # No zsh, no need for comletions
+    rm -rf usr/share/zsh/vendor-completions
 
-# systemd hw database manager (connman is in use)
-find usr etc -name '*systemd-hwdb*' -prune -exec rm -r {} \;
+    # drop gcc-6 python helpers
+    rm -rf usr/share/gcc-6
 
-# No need for fuse
-find usr etc -name '*fuse*' -prune -exec rm -r {} \;
+    # Drop sysvinit leftovers
+    rm -rf etc/init.d
+    rm -rf etc/rc[0-6S].d
 
-# lsb init function leftovers
-rm -rf usr/lib/lsb
+    # Drop upstart helpers
+    rm -rf etc/init
 
-# Utils using ncurses
-rm -f usr/bin/pg
-rm -f usr/bin/watch
-rm -f usr/bin/slabtop
+    # Various xtables helpers
+    rm -rf usr/lib/xtables
 
-# boot analyser
-rm -f usr/bin/systemd-analyze
+    # Drop all locales
+    # TODO: only remaining locale is actually "C". Should we really remove it?
+    rm -rf usr/lib/locale/*
 
-# Only needed when adding libraries
-rm -f usr/sbin/ldconfig*
+    # local compiler
+    rm -f usr/bin/localedef
 
-# Games, unused
-rm -rf usr/games
+    # lsb init function leftovers
+    rm -rf usr/lib/lsb
 
-# Unused systemd generators
-rm -f lib/systemd/system-generators/systemd-cryptsetup-generator
-rm -f lib/systemd/system-generators/systemd-debug-generator
-rm -f lib/systemd/system-generators/systemd-gpt-auto-generator
-rm -f lib/systemd/system-generators/systemd-hibernate-resume-generator
-rm -f lib/systemd/system-generators/systemd-rc-local-generator
-rm -f lib/systemd/system-generators/systemd-system-update-generator
-rm -f lib/systemd/system-generators/systemd-sysv-generator
+    # boot analyser
+    rm -f usr/bin/systemd-analyze
 
-# Efi blobs
-rm -rf usr/lib/systemd/boot
+    # Only needed when adding libraries
+    rm -f usr/sbin/ldconfig*
 
-# Translation catalogs
-rm -rf usr/lib/systemd/catalog
+    # Games, unused
+    rm -rf usr/games
 
-# Misc systemd utils
-rm -f usr/bin/bootctl
-rm -f usr/bin/busctl
-rm -f usr/bin/hostnamectl
-rm -f usr/bin/localectl
-rm -f usr/bin/systemd-cat
-rm -f usr/bin/systemd-cgls
-rm -f usr/bin/systemd-cgtop
-rm -f usr/bin/systemd-delta
-rm -f usr/bin/systemd-detect-virt
-rm -f usr/bin/systemd-mount
-rm -f usr/bin/systemd-path
-rm -f usr/bin/systemd-run
-rm -f usr/bin/systemd-socket-activate
-
-# Remove pam module to authenticate against a DB
-# plus libdb-5.3.so that is only used by this pam module
-rm -f usr/lib/*/security/pam_userdb.so
-rm -f usr/lib/*/libdb-5.3.so
-
-# remove NSS support for nis, nisplus and hesiod
-rm -f usr/lib/*/libnss_hesiod*
-rm -f usr/lib/*/libnss_nis*
+    # Remove pam module to authenticate against a DB
+    # plus libdb-5.3.so that is only used by this pam module
+    rm -f usr/lib/*/security/pam_userdb.so
+    rm -f usr/lib/*/libdb-5.3.so
+
+    # remove NSS support for nis, nisplus and hesiod
+    rm -f usr/lib/*/libnss_hesiod*
+    rm -f usr/lib/*/libnss_nis*
+}
+
+# Removing unneeded packages
+remove_packages()  {
```

Comment:
There's already a debos template parameter to provide a list of packages to remove `extra_packages_remove`.

Reply:
If I understand correctly, template used to remove non-essentials packages. With crush.sh critical packages like `apt` gets removed.  `crush.sh` invoked right after removing packages listed with `extra_packages_remove` .

Reply:
Oh I see, this function is only a local helper within the `crush.sh` script.  You made this function because we now have several places where we're calling that.  So yes, that's fine.